plugins/in_tail/tail_file.c

```diff
@@ -847,14 +866,15 @@ static int ml_flush_callback(struct flb_ml_parser *parser,
                                   file->mult_sbuf.size,
                                   &mult_buf, &mult_size);
 
-        flb_input_chunk_append_raw(ctx->ins,
-                                   file->tag_buf,
-                                   file->tag_len,
-                                   mult_buf,
-                                   mult_size);
+        ml_stream_buffer_append(file, mult_buf, mult_size);
         flb_free(mult_buf);
     }
 
+
+    if (mst->forced_flush) {
+        ml_stream_buffer_flush(ctx, file);
```

Comment:
Has this been tested with filters? The majority of them rewrite records and will truncate chunks to only include what they parse.

Reply:
@pwhelan that new `forced_flush` stuff it's disabled by default and only Tail uses it. So it's safe from that perspective.